Compilation failed.

CentOS 7.9

[ 50%] Building CXX object CMakeFiles/sdi2ddl.dir/src/sdi2ddl.cc.o
In file included from /usr/include/c++/4.8.2/unordered_map:35:0,
                 from /root/sdi2ddl-main/src/sdi2ddl.cc:3:
/usr/include/c++/4.8.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
In file included from /root/sdi2ddl-main/src/sdi2ddl.cc:4:0:
/root/sdi2ddl-main/src/include/collations.h:20:1: error: ‘unordered_map’ does not name a type
 unordered_map<int, collation_info> collation_map;
 ^
/root/sdi2ddl-main/src/include/collations.h: In function ‘bool find_collation(const int&, collation_info&)’:
/root/sdi2ddl-main/src/include/collations.h:34:8: error: ‘it’ does not name a type
   auto it = collation_map.find(id);
        ^
/root/sdi2ddl-main/src/include/collations.h:35:7: error: ‘it’ was not declared in this scope
   if (it == collation_map.end()) {
       ^
/root/sdi2ddl-main/src/include/collations.h:35:13: error: ‘collation_map’ was not declared in this scope
   if (it == collation_map.end()) {
             ^
/root/sdi2ddl-main/src/include/collations.h:38:15: error: ‘it’ was not declared in this scope
   collation = it->second;

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,13 @@ cmake_minimum_required(VERSION 3.16.3)
 
 project(sdi2ddl)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8")
+    message(FATAL_ERROR "Please use at least GCC version 4.8")
+  elseif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0")
+    set (CMAKE_CXX_FLAGS "-std=c++11")
+  endif()
+endif()
 
 MACRO (FIND_RAPIDJSON)
   MESSAGE("-- Searching for rapidjson/document.h")

